packages/ckeditor5-engine/src/model/element.js

```diff
@@ -209,6 +209,28 @@ export default class Element extends Node {
 		return node;
 	}
 
+	/**
+	 * Returns the parent element of the given name. Returns null if the element is not inside the desired parent.
+	 *
+	 * @param {String} parentName The name of the parent element to find.
+	 * @param {Object} [options] Options object.
+	 * @param {Boolean} [options.includeSelf=false] When set to `true` this node will be also included while searching.
+	 * @returns {module:engine/model/element~Element|null}
+	 */
+	findAncestor( parentName, options = { includeSelf: false } ) {
+		let parent = options.includeSelf ? this : this.parent;
+
+		while ( parent ) {
+			if ( parent.name === parentName ) {
```

Comment:
~`parent.is()` would be better.~

WDYT about using `parent.is()` here? Performance wise it shouldn't be a big impact now. Using `is()` is prefered in most cases but here we might need to check only for name. I don't think that `element.findAncestor( 'mode:element' )` is needed.

Reply:
I was thinking about more generic solution like [view element findAncestor()](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_view_element-Element.html#function-findAncestor) that accepts callback also (i.e., Matcher accepted patterns) but currently there is no use case for that scenario so I thought that we could extend this later if it will be needed.

Reply:
OK, maybe we'll change that anyway...